### ISSUE TITLE
Setoid example

### ISSUE THREAD

**Issue by chamini2** [hemanth/functional-programming-jargon#82]
The example of setoid uses equality between the elements of the array but it assumes there's an equality operator (`==`) between the elements of the setoid.

I would do the following:

```
Number.prototype.equals = function(num) {
  return num.valueOf() === this.valueOf();
}

(2).equals(2) // true
(2).equals(3) // false

Array.prototype.equals = function(arr) {
    var len = this.length
    if (len !== arr.length) {
        return false
    }
    for (var i = 0; i < len; i++) {
        if (! this[i].equals(arr[i])) {
            return false
        }
    }
    return true
}


[1,2].equals([1,2]) // true
[1,2].equals([1,2,3]) // false
[1,2].equals([2,3]) // false
```

**Comment by jethrolarson** [comment]
Yeah. It's too bad you have to keep defining your own equality function for everything in JS. Especially since it's generally considered a bad practice to modify native prototypes.

**Comment by chamini2** [comment]
I think that the examples are written in JS to make it easier to understand, but they're not the *best practices* nor intended to be. They're just to clarify any doubt left in the description of the functional programming term.

**Comment by jethrolarson** [comment]
I think the code we display should be moderately practical. It can be
incomplete or have uncovered cases. But people should be able to copy it

**Comment by chamini2** [comment]
Well, this is code that can be copied.
